### ISSUE TITLE
Missing inspector UI in post template block

### ISSUE THREAD

**Issue by draganescu** [WordPress/gutenberg#58469]
### Description

If you have a title in a post content block in a query loop, but the query has no results, and you select the post title in the list view, the inspector does not show any of the block’s controls, not does the block toolbar.

### Step-by-step reproduction instructions

- have some posts
- in the site editor
- in a template
- add a query loop
- add a filter that makes the query return no results
- open list view
- expand the contents of the query loop
- select the post title block
- notice missing inspector UI

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/107534/bf87e084-cd48-4a74-b87e-0145b3329c86

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
I was also able to reproduce this problem. Perhaps all the blocks inside the Post Template block will not have inspector UI displayed if there is no result of the loop. However, only block variations and block styles seem to be displayed.

![image](https://github.com/WordPress/gutenberg/assets/54422211/8dec5779-f4c1-4837-8fb1-abab65af6b4d)
![image](https://github.com/WordPress/gutenberg/assets/54422211/9355f6cf-0ff2-48fd-8e33-781cc067b50a)

**Comment by t-hamano** [comment]
My guess is that the underlying problem is that the block itself is not rendered to the canvas if there are no query results. Each block should have inserted its content into a slot via an `InspectorControl` within the Edit component or a `BlockEdit` hook. If the block itself, the Edit component, is not rendered, then the `InspectorControl` itself is not rendered either.

**Comment by draganescu** [comment]
Yeah I am not sure how to approach this best, I don't think it's a bug, but more a UI quick we need to account for.

**Comment by luminuu** [comment]
This behaviour also appears when you have no published posts in your install. Create a fresh install and remove the "Hello world!" post. If you then try to edit a query loop, this happens.

But shouldn't it be possible anyway to work with the blocks there to use as placeholders, even if no content is present?
